src/mixer/basetrackplayer.cpp

```diff
@@ -145,106 +139,134 @@ TrackPointer BaseTrackPlayerImpl::loadFakeTrack(bool bPlay, double filebpm) {
-                pLoopCue = m_pLoadedTrack->createAndAddCue();
-                pLoopCue->setType(Cue::LOOP);
+    // Clear loop
+    // It seems that the trick is to first clear the loop out point, and then
+    // the loop in point. If we first clear the loop in point, the loop out point
+    // does not get cleared.
+    m_pLoopOutPoint->set(-1);
+    m_pLoopInPoint->set(-1);
+
+    // The loop in and out points must be set here and not in slotTrackLoaded
+    // so LoopingControl::trackLoaded can access them.
+    const QList<CuePointer> trackCues(m_pLoadedTrack->getCuePoints());
+    QListIterator<CuePointer> it(trackCues);
+    while (it.hasNext()) {
+        CuePointer pCue(it.next());
+        if (pCue->getType() == Cue::LOOP) {
+            double loopStart = pCue->getPosition();
+            double loopEnd = loopStart + pCue->getLength();
+            qDebug() << "Found loop cue" << loopStart << (loopEnd - loopStart);
+            if (loopStart != -1 && loopEnd != -1 && loopStart <= loopEnd) {
+                qDebug() << "Loading loop cue" << loopStart << (loopEnd - loopStart);
+                m_pLoopInPoint->set(loopStart);
+                m_pLoopOutPoint->set(loopEnd);
+                break;
             }
-            pLoopCue->setPosition(loopStart);
-            pLoopCue->setLength(loopEnd - loopStart);
         }
+    }
 
-        // WARNING: Never. Ever. call bare disconnect() on an object. Mixxx
-        // relies on signals and slots to get tons of things done. Don't
-        // randomly disconnect things.
-        disconnect(m_pLoadedTrack.get(), 0, m_pBPM.get(), 0);
-        disconnect(m_pLoadedTrack.get(), 0, this, 0);
-        disconnect(m_pLoadedTrack.get(), 0, m_pKey.get(), 0);
-
-        // Do not reset m_pReplayGain here, because the track might be still
-        // playing and the last buffer will be processed.
+    connectLoadedTrack();
+}
 
-        m_pPlay->set(0.0);
+TrackPointer BaseTrackPlayerImpl::unloadTrack() {
+    if (!m_pLoadedTrack) {
+        // nothing to do
+        return TrackPointer();
     }
 
-    m_pLoadedTrack = pNewTrack;
-    if (m_pLoadedTrack) {
-        // Clear loop
-        // It seems that the trick is to first clear the loop out point, and then
-        // the loop in point. If we first clear the loop in point, the loop out point
-        // does not get cleared.
-        m_pLoopOutPoint->set(-1);
-        m_pLoopInPoint->set(-1);
-
-        // The loop in and out points must be set here and not in slotTrackLoaded
-        // so LoopingControl::trackLoaded can access them.
-        const QList<CuePointer> trackCues(pNewTrack->getCuePoints());
-        QListIterator<CuePointer> it(trackCues);
+    // Save the loops that are currently set in a loop cue. If no loop cue is
+    // currently on the track, then create a new one.
+    double loopStart = m_pLoopInPoint->get();
+    double loopEnd = m_pLoopOutPoint->get();
+    if (loopStart != -1 && loopEnd != -1 && loopStart <= loopEnd) {
```

Comment:
`-1` indicates no loop, correct? There is a constant for this called `kNoTrigger` in enginecontrol.h.